packages/passport-utils/src/passport-utils.ts

```diff
@@ -0,0 +1,158 @@
+import date from 'dayjs';
+import objectId from 'bson-objectid';
+import {IncomingMessage} from 'http';
+
+import {
+	PassportOauth20Profile as Profile,
+	PassportOauth20VerifyCallback as VerifyCallback
+} from './_passport-types';
+
+export type NewUserSessionProfile = {
+	id: string;
+	gid: string;
+	firstName: string;
+	lastName: string;
+	email: string;
+	isNew: true;
+	settings: string;
+};
+
+export type UserProfile = {
+	id: string;
+	isNew: boolean;
+} | {
+	school: string;
+	school_id: string; // eslint-disable-line camelcase
+};
+
+export interface BasicRequest extends IncomingMessage {
+	session: {
+		userProfile?: NewUserSessionProfile;
+	};
+	_table?: string;
+	_passportRedirect?: string;
+}
+
+export type BasicCallback<T = string> = (error?: Error, id?: T) => void;
+
+export function createProfileHandler(getUser: (gid: string, table: string) => Promise<UserProfile>) {
+	return async function handleProfile(
+		request: BasicRequest,
+		_: string,
+		__: string,
+		profile: Profile,
+		callback: VerifyCallback
+	): Promise<void> {
+		let user: object | NewUserSessionProfile;
+
+		try {
+			user = await getUser(profile.id, request._table);
+
+			if (user) {
+				callback(null, user);
+				return;
+			}
+		} catch (error) {
+			callback(error);
+			return;
+		}
+
+		const {id: gid, emails, displayName, name: {givenName: firstName, familyName: lastName}} = profile;
+		let firstNameFallback = displayName.slice(0, displayName.indexOf(' '));
+		let lastNameFallback = displayName.slice(displayName.indexOf(' ') + 1);
+
+		if (!firstNameFallback || !lastNameFallback) {
```

Comment:
I think if there is a displayName that has no spaces we should allow that to become first name fallback. 
```suggestion
		if (!firstNameFallback) {
```

Reply:
This logic is all yours once this is merged 😄 not going to make the change in this PR since this logic came straight from server.